hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -754,8 +755,10 @@ public String getRecordMergeStrategyId() {
    * Infers the merging behavior based on what the user sets (or doesn't set).
    * Validates that the user has not set an illegal combination of configs
    */
-  public static Triple<RecordMergeMode, String, String> inferCorrectMergingBehavior(RecordMergeMode recordMergeMode, String payloadClassName,
-                                                                                    String recordMergeStrategyId) {
+  public static Triple<RecordMergeMode, String, String> inferCorrectMergingBehavior(RecordMergeMode recordMergeMode,
```

Comment:
is there a good java docs on relation b/w merge mode and merge strategy id. it always confuses me.

Reply:
for eg, what does it mean if someone sets strategy id w/o setting merge mode. or are there conflicting values for merge mode and merge strategy that user can set

Reply:
Here's the documentation on merge mode and merge strategy ID: https://hudi.apache.org/docs/record_merger .  LMK if there's any improvement we can make.